### DISCUSSION TITLE
Error in Dax Studio when using Countrows

### DISCUSSION THREAD

**Discussion by Declan1984** [TabularEditor/TabularEditor3#395] [🙋♀️TE3 Q&A]
I spent a fair bit of time trying to troubleshoot an error with a simple COUNTROWS function when using Dax Studio.  When defining a measure like:

DEFINE
     test_measure = COUNTROWS('test_table')

EVALUATE
     test_measure

The query kept returning: "The expression specified in the query is not a valid table expression"

I found the solution here: https://community.powerbi.com/t5/DAX-Commands-and-Tips/The-expression-specified-in-the-query-is-not-a-valid-table/m-p/889744#M7817

The result needed to be wrapped like follows: 
	ROW("Result",test_measure)

Does anyone know why this is and does this only apply to COUNTROWS?

**Comment by otykier** [comment]
COUNTROWS returns a scalar value. The expression following EVALUATE must return a table value. For example, this doesn’t work either:

```
EVALUATE
1 + 2
```

But this works:

```
EVALUATE
{ 1 + 2 }
```

(Going forward, please keep the questions asked here on the topic of Tabular Editor).
